### PR TITLE
Adding 4.10->4.11 upgrade test for azure ipi workflows

### PR DESCRIPTION
**Pull request by shellyyang1989**

### CODE REVIEW

**Comment by shellyyang1989** [comment]
Summary of the failures:

1. The [job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30204/rehearse-30204-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-azure-ipi-disconnected-fullyprivate/1544886812982382592) failed at post step, not sure if it's a flake.

```
INFO[2022-07-07T07:01:58Z] Gathering network artifacts ...
[must-gather      ] OUT Using must-gather plug-in image: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9992ef35b4dc36401e0ea486acdeb4248489b44f4526d75a929815dbfb12c60
When opening a support case, bugzilla, or issue please include the following summary data along with any other requested information:
ClusterID: 53892f1a-66d9-470c-a268-c77773a6c57b
ClusterVersion: Stable at "4.11.0-0.nightly-2022-07-06-145812"
ClusterOperators:
	clusteroperator/kube-apiserver is not upgradeable because FeatureGatesUpgradeable: "TechPreviewNoUpgrade" does not allow updates
	clusteroperator/machine-config is not upgradeable because One or more machine config pools are updating, please see `oc get mcp` for further details
	clusteroperator/service-ca is progressing: Progressing: 
Progressing: service-ca does not have available replicas
[must-gather      ] OUT namespace/openshift-must-gather-r6mpj created
[must-gather      ] OUT clusterrolebinding.rbac.authorization.k8s.io/must-gather-58r55 created
Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (containers "gather", "copy" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "gather", "copy" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or containers "gather", "copy" must set securityContext.runAsNonRoot=true), seccompProfile (pod or containers "gather", "copy" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
[must-gather      ] OUT pod for plug-in image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e9992ef35b4dc36401e0ea4[86](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30204/rehearse-30204-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-azure-ipi-disconnected-fullyprivate/1544886812982382592#1:build-log.txt%3A86)acdeb42484[89](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30204/rehearse-30204-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-azure-ipi-disconnected-fullyprivate/1544886812982382592#1:build-log.txt%3A89)b44f4526d75a[92](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30204/rehearse-30204-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-azure-ipi-disconnected-fullyprivate/1544886812982382592#1:build-log.txt%3A92)[98](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30204/rehearse-30204-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-azure-ipi-disconnected-fullyprivate/1544886812982382592#1:build-log.txt%3A98)15dbfb12c60 created
[must-gather-dkfsv] OUT waiting for gather to complete
[must-gather-dkfsv] OUT gather never finished: pods "must-gather-dkfsv" not found
[must-gather      ] OUT clusterrolebinding.rbac.authorization.k8s.io/must-gather-58r55 deleted
[must-gather      ] OUT namespace/openshift-must-gather-r6mpj deleted
```

2. The [job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30204/rehearse-30204-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-azure-ipi-ovn/1544886811489210368) failed at installation. 


3. The rest 4 failed test is due to short of lease resources.

`ERRO[2022-07-07T05:34:09Z] error: Failed to acquire resource, current capacity: 0 free, 5 leased `

**Comment by shellyyang1989** [comment]
cc @jianlinliu @liangxia @jhou1 for review. Thanks!

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml [hunk 1]
```diff
@@ -127,7 +127,55 @@ tests:
...
+    env:
+      BASE_DOMAIN: qe.azure.devcluster.openshift.com
+      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.10 and @azure-ipi'
+      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.10 and @azure-ipi'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test
+    workflow: cucushift-installer-rehearse-azure-ipi-fips
+- as: azure-ipi-fullyprivate
+  cron: 0 1 */3 * *
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:167]: Maybe we can lower the frequency for this testing?

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:167]: Thanks for the suggestion! Does once a week sound good?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml [hunk 2]
```diff
@@ -127,7 +127,55 @@ tests:
...
+    cluster_profile: azure-qe
+    env:
+      BASE_DOMAIN: qe.azure.devcluster.openshift.com
+      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.10 and @azure-ipi and @disconnected'
+      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.10 and @azure-ipi and @disconnected'
+    test:
+    - ref: cucushift-installer-check
+    - chain: openshift-upgrade-qe-test-disconnected
+    workflow: cucushift-installer-rehearse-azure-ipi-disconnected-fullyprivate
+- as: azure-ipi-disconnected-fullyprivate-techpreview
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:142]: If I remember right, once techpreview feature is enabled, is not recommended to run minor version upgrade, should we only test it in z-stream release? I am thinking if we deserve it even in z-stream upgrade. If we really want to cover it, maybe give it low frequency.

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:142]: I was not aware of it. Thanks for letting me know. I'm fine to drop it. 
Btw, is it possible to add some content if upgrade is not supported in the documentation in the workflow or in our CI unifying spreadsheet? Either is fine.

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:142]: Let us drop it for now. I just noticed it this morning in https://coreos.slack.com/archives/CPJQCSN49/p1657231537322049.

I also added a new column named "upgrade supported" in our CI unifying spreadsheet

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml:167]: how about once 10 days.

Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml]: Sounds good to me. Updated. Thanks!

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml (hunk 2 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__nightly-4.11-upgrade-from-stable-4.10.yaml]: Thanks! Dropped the tests for techpreview.

**Pull request merged**